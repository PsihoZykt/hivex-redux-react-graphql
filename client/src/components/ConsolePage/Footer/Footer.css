.footer {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 20px 25px;
    background-color: white;
    border-top: 1px solid rgba(0, 0, 0, 0.2);
}

.footer__submit {
    font-weight: 500;
    font-size: 16px;
    line-height: 30px;
    background-image: linear-gradient(180deg, #45A6FF 0%, #0055FB 100%);
    background-position: 0 0, 100% 0;
    background-repeat: no-repeat;
    background-size: 100% 100%;
    background-clip: padding-box;
    border-radius: 5px;
    width: 110px;
    height: 40px;
    color: white;
    cursor: pointer;
    border: 2px solid transparent;


}

.footer__submit:hover {
    background-image: linear-gradient(0deg, rgba(255, 255, 255, 0.15), rgba(255, 255, 255, 0.15)), linear-gradient(180deg, #45A6FF 0%, #0055FB 100%);
    cursor: pointer;
}

.footer__submit:focus {
    border: 2px solid rgba(69, 165, 255, 0.5);
    border-radius: 5px;
    outline: none;

}

.footer__submit:active {
    background-image: linear-gradient(0deg, rgba(0, 0, 0, 0.15), rgba(0, 0, 0, 0.15)), linear-gradient(180deg, #45A6FF 0%, #0055FB 100%);
    border: 2px solid transparent;
}

.footer__format {
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 10px;
    cursor: pointer;
    padding: 5px;
}

.footer__format:hover {

}

.footer__format:focus {
    border: 2px solid rgba(178, 165, 255, 0.5);
    filter: invert(19%) sepia(86%) saturate(4269%) hue-rotate(218deg) brightness(98%) contrast(105%);

    border-radius: 7px;
}