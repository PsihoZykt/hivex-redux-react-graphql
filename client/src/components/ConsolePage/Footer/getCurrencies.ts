import {gql} from "@apollo/client";
import {GraphQLCurrencyFieldType} from "types/EntityTypes/GraphQLCurrencyFieldType";
import {GraphqlAnyEntityFieldType} from "types/EntityTypes/EntityTypes";

export const getCurrenciesQuery = gql`query GetCurrencies {
    getCurrencies {
        _id
        name
        code
    }
}`
export const createGetCurrenciesQuery = <T extends GraphqlAnyEntityFieldType>(fields: T[]) => {
    try {
        return gql`query GetCurrencies($input: CurrencyFilter) {
            getCurrencies(input: $input) {
                ${fields}
            }
        }`
    } catch (e: any) {
        throw new Error(e)
    }

}

export const ADD_CURRENCY = gql`mutation AddCurrency($input: CurrencyInput) {
    addCurrency(input: $input) {
        name, _id
    }
}`
export const DELETE_CURRENCIES = gql`mutation DeleteCurrencies($input: CurrencyFilter){
    deleteCurrencies(input: $input){
        name, _id
    }
}`
export const UPDATE_CURRENCIES = gql`mutation UpdateCurrencies($input: UpdateCurrencyInput){
    updateCurrencies(input: $input){
        name, _id
    }
}`
