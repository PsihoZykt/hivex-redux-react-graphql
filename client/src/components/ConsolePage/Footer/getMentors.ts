import {gql} from "@apollo/client";
import {GraphQLMentorFieldType} from "types/EntityTypes/GraphQLMentorFieldType";
import {GraphqlAnyEntityFieldType} from "types/EntityTypes/EntityTypes";

export const getMentorsQuery = gql`query GetMentors {
    getMentors {
        _id
        name
        level
        workDuration
        salary
        country
        timestamp
        techStack
    }
}`
export const createGetMentorsQuery = <T extends GraphqlAnyEntityFieldType>(fields: T[]) => {
    try {
        return gql`query GetMentors($input: MentorFilter) {
            getMentors(input: $input) {
                ${fields}
            }
        }`
    } catch (e: any) {
        throw new Error(e)
    }

}

export const ADD_MENTOR = gql`mutation AddMentor($input: MentorInput) {
    addMentor(input: $input) {
        name, _id
    }
}`
export const DELETE_MENTORS = gql`mutation DeleteMentors($input: MentorFilter){
    deleteMentors(input: $input){
        name, _id
    }
}`
export const UPDATE_MENTORS = gql`mutation UpdateMentors($input: UpdateMentorInput){
    updateMentors(input: $input){
        name, _id
    }
}`
