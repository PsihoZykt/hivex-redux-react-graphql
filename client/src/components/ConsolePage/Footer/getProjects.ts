import {gql} from "@apollo/client";
import {GraphQLProjectFieldType} from "types/EntityTypes/GraphQLProjectFieldType";
import {GraphqlAnyEntityFieldType} from "types/EntityTypes/EntityTypes";

export const getProjectsQuery = gql`query GetProjects {
  getProjects {
      _id
    name
    status
    mentor {
        _id
      salary
      workDuration
      level
      name
      country
      timestamp
      techStack
    }
    country
    duration
    timestamp
    techStack
  }
}`
export const createGetProjectsQuery = <T extends GraphqlAnyEntityFieldType>(fields: T[]) => {
    try {
        return gql`query GetProjects($input: ProjectFilter) {
            getProjects(input: $input) {
                ${fields}
            }
        }`
    } catch (e: any) {
      throw new Error(e)
    }

}

export const ADD_PROJECT = gql`mutation GetProject($input: ProjectInput) {
    addProject(input: $input) {
        name, _id
    }
}`
export const DELETE_PROJECTS = gql`mutation DeleteProjects($input: ProjectFilter){
    deleteProjects(input: $input){
        name, _id
    }
}`
export const UPDATE_PROJECTS = gql`mutation UpdateProjects($input: UpdateProjectInput){
    updateProjects(input: $input){
        name, _id
    }
}`
