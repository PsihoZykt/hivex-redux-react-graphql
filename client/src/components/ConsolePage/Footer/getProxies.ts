import {gql} from "@apollo/client";
import {GraphQLProxyFieldType} from "types/EntityTypes/GraphQLProxyFieldType";
import {GraphqlAnyEntityFieldType} from "types/EntityTypes/EntityTypes";

export const getProxiesQuery = gql`query GetUsers {
    getProxies {
        _id
        name
        country
        timestamp
        bank
        currency {
            _id
            name
            code
        }
    }
}`
export const createGetProxiesQuery = <T extends GraphqlAnyEntityFieldType>(fields: T[]) => {
    try {
        return gql`query GetProxies($input: ProxyFilter) {
            getProxies(input: $input) {
                ${fields}
            }
        }`
    } catch (e: any) {
       throw new Error(e)
    }

}

export const ADD_PROXY = gql`mutation GetProxies($input: ProxyInput) {
    addProxy(input: $input) {
        name, _id
    }
}`
export const DELETE_PROXIES = gql`mutation DeleteProxies($input: ProxyFilter){
    deleteProxies(input: $input){
        name, _id
    }
}`
export const UPDATE_PROXIES = gql`mutation UpdateProxies($input: UpdateProxyInput){
    updateProxies(input: $input){
        name, _id
    }
}`
