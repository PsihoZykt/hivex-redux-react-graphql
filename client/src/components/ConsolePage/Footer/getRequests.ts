import {gql} from "@apollo/client";
import {GraphQLRequestFieldType} from "types/EntityTypes/GraphQLRequestFieldType";
import {GraphqlAnyEntityFieldType} from "types/EntityTypes/EntityTypes";

export const getRequestsQuery = gql`query GetRequests {
    getRequests {
        _id
        request
        response
        createdAt
    }
}`
export const createGetRequestsQuery = <T extends GraphqlAnyEntityFieldType>(fields: T[]) => {
    try {
        return gql`query GetRequests($input: RequestFilter) {
            getRequests(input: $input) {
                ${fields}
            }
        }`
    } catch (e: any) {
        throw  new Error(e)
    }

}

export const ADD_REQUEST = gql`mutation GetRequests($input: RequestInput) {
    addRequest(input: $input) {
        _id, createdAt,request,response
    }
}`
export const DELETE_REQUESTS = gql`mutation DeleteRequests($input: RequestFilter){
    deleteRequests(input: $input){
        _id, createdAt,request,response

    }
}`
export const UPDATE_REQUESTS = gql`mutation UpdateRequests($input: UpdateRequestInput){
    updateRequests(input: $input){
        _id, createdAt,request,response

    }
}`
