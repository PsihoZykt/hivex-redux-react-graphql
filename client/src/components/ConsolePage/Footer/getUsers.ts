import {gql} from "@apollo/client";

export const getUsersQuery =
  gql`query GetUsers {
      getUsers {
          _id
          name
          proxy {
              _id
              name
              timestamp
              country
              bank
              currency {
                  _id
                  name
                  code
              }
          }
          salary
          project {
              _id
              name
              status
              mentor {
                  _id
                  salary
                  workDuration
                  level
                  name
                  country
                  timestamp
                  techStack
              }
              country
              duration
                    timestamp
                    techStack
                }
                workDuration
                level
                techStack
                timeStamp
            }
        }`

export const createGetUsersQuery = ({body, queryString = {}}: any) => {
    console.log(queryString)
    try {
        return gql`query GetUsers {
            getUsers(filter: ${queryString}) {
                ${body}
            }
        }`
    } catch (e: any) {
        console.log(e)
    }

}

export const ADD_USER = gql`mutation GetUsers($input: UserInput) {
    addUser(input: $input) {
        name, _id
    }
}`
export const DELETE_USERS = gql`mutation DeleteUsers($input: UserFilter){
    deleteUser(input: $input){
        name, _id
    }
}`
export const UPDATE_USERS = gql`mutation UpdateUsers($input: UpdateUsersInput){
    updateUsers(input: $input){
        name, _id
    }
}`