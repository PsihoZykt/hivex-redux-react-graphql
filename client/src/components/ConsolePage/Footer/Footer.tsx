import GithubLink from 'common/GithubLink/GithubLink'
import format from 'assets/img/consolePage/format.svg'
import React, {Dispatch, useState} from 'react'
import './Footer.css'
import {useLazyQuery, useMutation, useQuery} from "@apollo/client";
import {
  ADD_USER,
  createGetUsersQuery,
  DELETE_USERS,
  getUsersQuery,
  UPDATE_USERS,
} from "components/ConsolePage/Footer/getUsers";
import {
  getAddEntityMutation,
  getDeleteEntityMutation,
  getQuery,
  getUpdateEntityMutation,
  getValues
} from "helpers/graphql/graphqlHelper";
import {getFormattedJSON} from "helpers/json/format";
import {
  ADD_CURRENCY,
  createGetCurrenciesQuery,
  DELETE_CURRENCIES,
  getCurrenciesQuery,
  UPDATE_CURRENCIES
} from "components/ConsolePage/Footer/getCurrencies";
import {
  ADD_MENTOR,
  createGetMentorsQuery,
  DELETE_MENTORS,
  getMentorsQuery,
  UPDATE_MENTORS
} from "components/ConsolePage/Footer/getMentors";
import {
  ADD_PROXY,
  createGetProxiesQuery,
  DELETE_PROXIES,
  getProxiesQuery,
  UPDATE_PROXIES
} from "components/ConsolePage/Footer/getProxies";
import {
  ADD_REQUEST,
  createGetRequestsQuery,
  DELETE_REQUESTS,
  getRequestsQuery,
  UPDATE_REQUESTS
} from "components/ConsolePage/Footer/getRequests";
import {
  ADD_PROJECT,
  createGetProjectsQuery,
  DELETE_PROJECTS,
  getProjectsQuery,
  UPDATE_PROJECTS
} from "components/ConsolePage/Footer/getProjects";
import {GraphQLUserFieldType} from "types/EntityTypes/GraphQLUserFieldType";
import {GraphqlAnyEntityFieldType} from "types/EntityTypes/EntityTypes";
import {GraphQLCurrencyFieldType} from "types/EntityTypes/GraphQLCurrencyFieldType";

type FooterProps = {
  request: String
  setResponse: Dispatch<string>
}

export enum CommandNames {
  GET_FIELDS = "get-fields",
  GET_USERS = "get-users",
  ADD_USER = "add-user",
  DELETE_USERS = "delete-users",
  UPDATE_USERS = "update-users",

  GET_CURRENCIES = "get-currencies",
  ADD_CURRENCY = "add-currency",
  DELETE_CURRENCIES = "delete-currencies",
  UPDATE_CURRENCIES = "update-currencies",

  GET_MENTORS = "get-mentors",
  ADD_MENTOR = "add-mentors",
  DELETE_MENTORS = "delete-mentors",
  UPDATE_MENTORS = "update-mentors",

  GET_PROJECTS = "get-projects",
  ADD_PROJECT = "add-project",
  DELETE_PROJECTS = "delete-projects",
  UPDATE_PROJECTS = "update-projects",

  GET_PROXIES = "get-proxies",
  ADD_PROXY = "add-proxy",
  DELETE_PROXIES = "delete-proxies",
  UPDATE_PROXIES = "update-proxies",

  GET_REQUESTS = "get-requests",
  ADD_REQUEST = "add-request",
  DELETE_REQUESTS = "delete-requests",
  UPDATE_REQUESTS = "update-requests",

}

export enum Fields {
  USERS = 'users',
  CURRENCIES = "currencies",
  MENTORS = "mentors",
  PROXIES = "proxies",
  REQUESTS = "requests",
  PROJECTS = "projects"
}


export type CommandType = {
  prefix: "hivex"
  commandName: CommandNames
  valuesKey: "-values"
}
export const Footer = ({request, setResponse}: FooterProps) => {

  const [queryFieldsArr, setQueryFieldsArr] = useState<GraphqlAnyEntityFieldType[]>(["name"])
  const [getUsers, getUsersData] = useLazyQuery(createGetUsersQuery(queryFieldsArr), {
    errorPolicy: "ignore", onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })


  const [getCurrencies, getCurrenciesData] = useLazyQuery(createGetCurrenciesQuery(queryFieldsArr), {
    errorPolicy: "ignore", onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })

  const [getMentors, getMentorsData] = useLazyQuery(createGetMentorsQuery(queryFieldsArr), {
    errorPolicy: "ignore", onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [getProjects, getProjectsData] = useLazyQuery(createGetProjectsQuery(queryFieldsArr), {
    errorPolicy: "ignore", onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [getProxies, getProxiesData] = useLazyQuery(createGetProxiesQuery(queryFieldsArr), {
    errorPolicy: "ignore", onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [getRequests, getRequestsData] = useLazyQuery(createGetRequestsQuery(queryFieldsArr), {
    errorPolicy: "ignore", onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })


  const usersData = useQuery(getUsersQuery)
  const currenciesData = useQuery(getCurrenciesQuery)
  const mentorsData = useQuery(getMentorsQuery)
  const proxiesData = useQuery(getProxiesQuery)
  const requestsData = useQuery(getRequestsQuery)
  const projectsData = useQuery(getProjectsQuery)
  const [addUser, addUserData] = useMutation(ADD_USER, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [addCurrency, addCurrencyData] = useMutation(ADD_CURRENCY, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })

  const [addMentor, addMentorData] = useMutation(ADD_MENTOR, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })

  const [addProject, addProjectData] = useMutation(ADD_PROJECT, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [addProxy, addProxyData] = useMutation(ADD_PROXY, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })


  const [addRequest, addRequestData] = useMutation(ADD_REQUEST, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [deleteUsers, deleteUsersData] = useMutation(DELETE_USERS, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [deleteCurrencies, deleteCurrenciesData] = useMutation(DELETE_CURRENCIES, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [deleteMentors, deleteMentorsData] = useMutation(DELETE_MENTORS, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [deleteProjects, deleteProjectsData] = useMutation(DELETE_PROJECTS, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [deleteProxies, deleteProxiesData] = useMutation(DELETE_PROXIES, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [deleteRequests, deleteRequestData] = useMutation(DELETE_REQUESTS, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })


  const [updateUsers, updateUsersData] = useMutation(UPDATE_USERS, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [updateCurrencies, updateCurrenciesData] = useMutation(UPDATE_CURRENCIES, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [updateMentors, updateMentorsData] = useMutation(UPDATE_MENTORS, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [updateProjects, updateProjectsData] = useMutation(UPDATE_PROJECTS, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [updateProxies, updateUProxiesData] = useMutation(UPDATE_PROXIES, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })
  const [updateRequests, updateRequestsData] = useMutation(UPDATE_REQUESTS, {
    onCompleted: (data) => {
      setResponse(JSON.stringify(data, null, "\t"))
    }
  })


  const onSubmitRequest = async () => {
    let response = ""
    const requestParts = request.split(' ')
    let prefix = requestParts[0] as "hivex";
    let commandName = requestParts[1] as unknown as CommandNames
    let valuesKey = requestParts[2] as "-values"
    let command: CommandType = {
      prefix, commandName, valuesKey,
    }
    try {
      if (command.prefix === "hivex") {

        if (command.commandName === CommandNames.GET_FIELDS) {
          const fields = getValues(request)
          if (fields.includes(Fields.PROJECTS)) {
            response += getFormattedJSON(JSON.stringify(projectsData.data)) + "\n"
          }
          if (fields.includes(Fields.USERS)) {
            response += getFormattedJSON(JSON.stringify(usersData.data))
          }
          if (fields.includes(Fields.CURRENCIES)) {
            response += getFormattedJSON(JSON.stringify(currenciesData.data))
          }
          if (fields.includes(Fields.PROXIES)) {
            response += getFormattedJSON(JSON.stringify(proxiesData.data))
          }
          if (fields.includes(Fields.REQUESTS)) {
            response += getFormattedJSON(JSON.stringify(requestsData.data))
          }
          if (fields.includes(Fields.MENTORS)) {
            response += getFormattedJSON(JSON.stringify(mentorsData.data))
          }
          setResponse(response)
        } else if (command.valuesKey === "-values") {
          if (command.commandName === CommandNames.GET_USERS) {
            const query = getQuery<GraphQLUserFieldType>(request)
            setQueryFieldsArr(query.fields)
            await getUsers({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.GET_CURRENCIES) {
            const query = getQuery<GraphQLCurrencyFieldType>(request)
            setQueryFieldsArr(query.fields)
            await getCurrencies({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.GET_MENTORS) {
            const query = getQuery(request)

            setQueryFieldsArr( query.fields)
            await getMentors({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.GET_PROJECTS) {
            const query = getQuery(request)
            setQueryFieldsArr(query.fields)
            await getProjects({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.GET_PROXIES) {
            const query = getQuery(request)
            setQueryFieldsArr(query.fields)
            await getProxies({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.GET_REQUESTS) {
            const query = getQuery(request)
            setQueryFieldsArr( query.fields)
            await getRequests({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.ADD_USER) {
            const mutation = getAddEntityMutation(request)
            await addUser({variables: {input: mutation.filter}})
          } else if (command.commandName === CommandNames.ADD_CURRENCY) {
            const mutation = getAddEntityMutation(request)
            await addCurrency({variables: {input: mutation.filter}})
          } else if (command.commandName === CommandNames.ADD_MENTOR) {
            const mutation = getAddEntityMutation(request)
            await addMentor({variables: {input: mutation.filter}})
          } else if (command.commandName === CommandNames.ADD_PROJECT) {
            const mutation = getAddEntityMutation(request)
            await addProject({variables: {input: mutation.filter}})
          } else if (command.commandName === CommandNames.ADD_PROXY) {
            const mutation = getAddEntityMutation(request)
            await addProxy({variables: {input: mutation.filter}})
          } else if (command.commandName === CommandNames.ADD_REQUEST) {
            const mutation = getAddEntityMutation(request)
            await addRequest({variables: {input: mutation.filter}})
          } else if (command.commandName === CommandNames.DELETE_USERS) {
            const query = getDeleteEntityMutation(request)
            await deleteUsers({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.DELETE_CURRENCIES) {
            const query = getDeleteEntityMutation(request)
            await deleteCurrencies({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.DELETE_MENTORS) {
            const query = getDeleteEntityMutation(request)
            await deleteMentors({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.DELETE_PROJECTS) {
            const query = getDeleteEntityMutation(request)
            await deleteProjects({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.DELETE_PROXIES) {
            const query = getDeleteEntityMutation(request)
            await deleteProxies({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.DELETE_REQUESTS) {
            const query = getDeleteEntityMutation(request)
            await deleteRequests({variables: {input: query.filter}})
          } else if (command.commandName === CommandNames.UPDATE_USERS) {
            let mutation = getUpdateEntityMutation(request)
            console.log(mutation.filter)
            await updateUsers({variables: {input: {filter: mutation.filter, set: mutation.set}}})
          } else if (command.commandName === CommandNames.UPDATE_MENTORS) {
            let mutation = getUpdateEntityMutation(request)
            console.log(mutation.filter)
            await updateMentors({variables: {input: {filter: mutation.filter, set: mutation.set}}})
          } else if (command.commandName === CommandNames.UPDATE_PROJECTS) {
            let mutation = getUpdateEntityMutation(request)
            console.log(mutation.filter)
            await updateProjects({variables: {input: {filter: mutation.filter, set: mutation.set}}})
          } else if (command.commandName === CommandNames.UPDATE_PROXIES) {
            let mutation = getUpdateEntityMutation(request)
            console.log(mutation.filter)
            await updateProxies({variables: {input: {filter: mutation.filter, set: mutation.set}}})
          } else if (command.commandName === CommandNames.UPDATE_REQUESTS) {
            let mutation = getUpdateEntityMutation(request)
            console.log(mutation.filter)
            await updateRequests({variables: {input: {filter: mutation.filter, set: mutation.set}}})
          } else if (command.commandName === CommandNames.UPDATE_CURRENCIES) {
            let mutation = getUpdateEntityMutation(request)
            console.log(mutation.filter)
            await updateCurrencies({variables: {input: {filter: mutation.filter, set: mutation.set}}})
          }
        }
      } else {
        setResponse(`
      Incorrect command: 
      command syntax: 
      [hivex commandName -values fieldOne -f? filterOne? | field2 -f? filterTwo?] (? means optional) 
      your command:
      ${request}
      ${command.prefix === "hivex" ? "" : "command should start with hivex"}
      ${command.commandName === CommandNames.GET_USERS || command.commandName === CommandNames.GET_FIELDS ? "" : `Incorrect commandName. Supported commands are: ${CommandNames.GET_USERS}, ${CommandNames.GET_FIELDS}`}
      ${command.valuesKey === "-values" ? "" : "Write -values after commandName"}
      `)
      }
    } catch (e) {
      console.log(e)
    }
  }


  return (
    <div className="footer">
      <button
        onClick={() => onSubmitRequest()}
        className="footer__submit"
      >
        Отправить
      </button>
      <GithubLink/>
      <div
        tabIndex={0}
        className="footer__format"
        onClick={() => {
        }}
      >
        <img src={format} alt="some rectangles with different width"/>
        Форматировать
        {usersData.loading && <p>Loading...</p>}
        {usersData.error && <p>Error!</p>}

      </div>

    </div>
  )
}

export default Footer