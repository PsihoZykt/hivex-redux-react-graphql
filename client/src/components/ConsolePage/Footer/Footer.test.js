import TestRenderer from "react-test-renderer";

import {MockedProvider} from "@apollo/client/testing";

import {Footer} from "./Footer";
import {expect, it} from "@jest/globals";
import {createGetUsersQuery} from "./graphqlQueries/getUsers.graphql";
import '@testing-library/jest-dom'
import {gql, useQuery} from "@apollo/client";
import {getQuery} from "../../../helpers/graphql/graphqlHelper";
import {createGetProjectsQuery} from "./graphqlQueries/getProjects.graphql";
import {act, render} from "@testing-library/react";
import {createGetCurrenciesQuery} from "./graphqlQueries/getCurrencies.graphql";
import {createGetMentorsQuery} from "./graphqlQueries/getMentors.graphql";
import {createGetProxiesQuery} from "./graphqlQueries/getProxies.graphql";
import {createGetRequestsQuery} from "./graphqlQueries/getRequests.graphql";
import {useGetUsers} from "./getUsers";

const mocks = []; // We'll fill this in next

it("renders without error", () => {
    const {getByText} = render(
        <MockedProvider>
            <Footer/>
        </MockedProvider>
    );

    expect(getByText("Loading...")).toBeInTheDocument()


});
it("should render loading state initially", () => {


    const user = {name: "1"};
    const mocks = [

        {

            request: {

                query: createGetUsersQuery(["name", "_id"]),

                variables: {input: {name: "123"}},

            },

            result: {data: user},

        },

    ];
    let response

    const component = TestRenderer.create(
        <MockedProvider mocks={null} addTypename={false}>

            <Footer setResponse={(body) => response = body} request={"dsa"}/>

        </MockedProvider>
    );
    // find the button and simulate a click
    //
    const button = component.root.findByType("button");
    button.props.onClick(); // fires the mutation
    expect(response).toMatch(/Incorrect command/);

});
describe("all get-commands should work", () => {

    test("createGetUsersQuery with name should create correct queries", () => {
        let input = "hivex get-users -values name"
        let queryObj = getQuery(input)
        const query = createGetUsersQuery(queryObj.fields)
        expect(query).toEqual(
            gql`query GetUsers($input: UserFilter) {
                getUsers(input: $input) {
                    name
                }
            }`)
    });
    test("createGetUsersQuery with name project.name should create correct queries", () => {
        let input = "hivex get-users -values name | project.name"
        let queryObj = getQuery(input)
        const query = createGetUsersQuery(queryObj.fields)
        expect(query).toEqual(gql`query GetUsers($input: UserFilter)
        { getUsers(input: $input) {
            name project{name}
        }
        }`)
    });
    test("createGetUsersQuery with _id name project{name mentor{name}} should create correct queries", async () => {
        const user = {name: "1"};
        const mocks = [

            {

                request: {

                    query: createGetUsersQuery(["name", "_id"]),

                    variables: {input: {name: "123"}},

                },

                result: {data: user},

            },

        ];
        let input = "hivex get-users -values _id | name | project.name | project.mentor.name"
        const component = TestRenderer.create(
            <MockedProvider mocks={mocks} addTypename={false}>

                <Footer setResponse={(body) => console.log(body)} request={input}/>

            </MockedProvider>
        );

       const mock =  jest.mock('./getUsers')
        const button = component.root.findByType("button");
        button.props.onClick(); // fires the mutation
        expect(mock).toHaveBeenCalled()



        // expect(response).toMatch(/Incorrect command/);
        // let queryObj = getQuery(input)
        // const query = createGetUsersQuery(queryObj.fields)
        //
        // console.log(query)
        // expect(query).toEqual(gql`query GetUsers($input: UserFilter)
        // { getUsers(input: $input) {
        //     _id name project{name mentor {
        //         name
        //     }}
        // }
        // }`)
    });


    test('createGetProjectsQuery should create correct query', () => {

        let input = "hivex get-projects -values name"
        let queryObj = getQuery(input)
        const query = createGetProjectsQuery(queryObj.fields)
        expect(query).toEqual(gql`query GetProjects($input: ProjectFilter) { getProjects(input: $input) {
            name
        }
        }`)
    })
    test('createGetCurrenciesQuery should create correct query', () => {

        let input = "hivex get-currencies -values name"
        let queryObj = getQuery(input)
        const query = createGetCurrenciesQuery(queryObj.fields)
        expect(query).toEqual(gql`query GetCurrencies($input: CurrencyFilter) { getCurrencies(input: $input) {
            name
        }
        }`)
    })
    test('createGetMentorsQuery should create correct query', () => {

        let input = "hivex get-mentors -values name"
        let queryObj = getQuery(input)
        const query = createGetMentorsQuery(queryObj.fields)
        expect(query).toEqual(gql`query GetMentors($input: MentorFilter) { getMentors(input: $input) {
            name
        }
        }`)
    })
    test('createGetProxyQuery should create correct query', () => {

        let input = "hivex get-proxies -values name"
        let queryObj = getQuery(input)
        const query = createGetProxiesQuery(queryObj.fields)
        expect(query).toEqual(gql`query GetProxies($input: ProxyFilter) { getProxies(input: $input) {
            name
        }
        }`)
    })
    test('createGetRequestsQuery should create correct query', () => {
        let input = "hivex get-requests -values response"
        let queryObj = getQuery(input)
        const query = createGetRequestsQuery(queryObj.fields)
        expect(query).toEqual(gql`query GetRequests($input: RequestFilter) { getRequests(input: $input) {
            response
        }
        }`)
    })



})