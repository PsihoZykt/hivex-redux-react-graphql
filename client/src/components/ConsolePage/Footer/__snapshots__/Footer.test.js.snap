// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders top project list 1`] = `
<div>
  <div
    class="footer"
  >
    <button
      class="footer__submit"
    >
      Отправить
    </button>
    <div>
      <div
        class="github-link"
      >
        https://github.com/PsihoZykt/API-console
      </div>
    </div>
    <div
      class="footer__format"
      tabindex="0"
    >
      <img
        alt="some rectangles with different width"
        src="format.svg"
      />
      Форматировать
      <p>
        Loading...
      </p>
    </div>
  </div>
</div>
`;
